src/transports/tcp/TcpConnectionAcceptor.cpp

```diff
@@ -114,4 +114,12 @@ void TcpConnectionAcceptor::stop() {
       [this] { serverSocket_.reset(); });
   serverThread_.reset();
 }
+
+folly::Optional<uint16_t> TcpConnectionAcceptor::listeningPort() const {
+  if (!serverSocket_) {
+    return folly::none;
```

Comment:
ConnectionAcceptor's constructor accepts Options parameter. This struct has a "port" value in it. 
So if we don't have serverSocket_, I believe we should provide this "port" value.
Perhaps, we should not ask for serverSocket_->getAddress().getPort() at all if the port value in the options parameter is not 0.